presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/api/CoordinatorConfigAPI.java

```diff
@@ -0,0 +1,158 @@
+import org.apache.logging.log4j.Logger;
+
+import javax.inject.Singleton;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import static com.teradata.prestomanager.common.RequestBuilder.newRequestBuilder;
+
+@Path("/coordinator/config")
+@Api(description = "the config API for coordinator")
+@Singleton
+public final class CoordinatorConfigAPI
+{
+    private static final Logger LOGGER = LogManager.getLogger(CoordinatorConfigAPI.class);
+
+    @GET
+    @Produces({"text/plain"})
+    @ApiOperation(value = "Get available configuration files")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Retrieved configuration", response = String.class)})
+    public synchronized Response getConfig()
+    {
+        try {
+            if (NodeSet.getCoordinator().size() != 1) {
+                throw new Exception("More than one coordinator existed");
+            }
+        }
+        catch (Exception e) {
+            LOGGER.error("More than one coordinator existed", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+        }
+        return newRequestBuilder(Iterables.get(NodeSet.getCoordinator(), 0)
+                .resolve("/config")).buildGet().invoke();
+    }
+
+    @GET
+    @Path("/{file}")
+    @Produces({"text/plain"})
+    @ApiOperation(value = "Get configuration by file")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Retrieved file", response = String.class),
+            @ApiResponse(code = 404, message = "Resource not found")})
+    public synchronized Response getConfigFile(
+            @PathParam("file") @ApiParam("The name of a file") String file)
+    {
+        try {
+            if (NodeSet.getCoordinator().size() != 1) {
+                throw new Exception("More than one coordinator existed");
+            }
+        }
+        catch (Exception e) {
+            LOGGER.error("More than one coordinator existed", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+        }
+        return newRequestBuilder(Iterables.get(NodeSet.getCoordinator(), 0)
+                .resolve("/config/" + file)).buildGet().invoke();
+    }
+
+    @POST
+    @Path("/{file}")
+    @ApiOperation(value = "Replace this file with the file at the given URL")
+    @ApiResponses(value = {
+            @ApiResponse(code = 202, message = "Acknowledged request"),
+            @ApiResponse(code = 409, message = "Request conflicts with current state")})
+    public synchronized Response setConfigFileByURL(
+            @PathParam("file") @ApiParam("The name of a file") String file,
+            String url)
+    {
+        try {
+            if (NodeSet.getCoordinator().size() != 1) {
```

Comment:
You should store the result of `getCoordinator`. Otherwise, the result of this test may change before you call it again.